vendor/src/github.com/docker/distribution/registry/client/auth/session.go

```diff
@@ -151,6 +155,19 @@ func (rs RepositoryScope) String() string {
 	return fmt.Sprintf("repository:%s:%s", rs.Repository, strings.Join(rs.Actions, ","))
 }
 
+// RegistryScope represents a token scope for access
+// to resources in the registry.
+type RegistryScope struct {
```

Comment:
This particular type was just added with https://github.com/docker/distribution/commit/145abeea7bb8a6bfe54d9cf05222caf9b8f7b80d.

Other areas in the code were just using `auth.Resource` for checking, https://github.com/docker/distribution/blob/641f1029677e49faa277f7769797518e973865fd/registry/handlers/app.go#L867. We want something with a stronger type for creating the scopes fro clients.